wait-for.c

```diff
@@ -245,21 +243,21 @@ int main(int argc, const char **argv) {
 		goto exit;
 	}
 
-	if (args.username == NULL) {
-		// attempt to populate based on current UID
-		struct passwd *pwd = getpwuid(getuid());
-		if (pwd == NULL) {
+	errno = 0;
+	if (args.username) {
+		pwd = getpwnam(args.username);
+	} else {
+		pwd = getpwuid(getuid());
+	}
+
+	if (pwd == NULL) {
+		if (errno) {
 			perror("could not get passwd entry for the user");
 			status = 1;
-			goto exit;
+		} else {
+			fputs("unknown user\n", stderr);
```

Comment:
- Prefix with `error:`
- Use `printf` so you can supply the username that was erroneous (`unknown user: %s`)
- Make sure to exit if there's an error.